server/controllers/auth.controller.ts

```diff
@@ -0,0 +1,116 @@
+const bcrypt = require('bcryptjs')
+const { validationResult } = require('express-validator')
+const jwt = require('jsonwebtoken')
+const User = require('../models/User')
+const { SALT_LENGTH, JWT_TOKEN_LIFE_TIME } = require('../config/constants')
+
+class AuthController {
+    static generateAccessToken(id) {
+        const payload = {
+            id,
+        }
+
+        return jwt.sign(payload, process.env.SECRET_KEY, {
+            expiresIn: JWT_TOKEN_LIFE_TIME,
+        })
+    }
+
+    static async hashPassword(password) {
+        return bcrypt.hash(password, SALT_LENGTH)
+    }
+
+    async post(req, res) {
```

Comment:
why some of the methods are static but others aren't static?

Reply:
In static methods I have defined functions used as helper.